S/SPIRV_LLVM_Translator_unified/build_tarballs.jl

```diff
@@ -0,0 +1,118 @@
+using BinaryBuilder, Pkg
+using Base.BinaryPlatforms
+
+include("../../fancy_toys.jl")
+
+name = "SPIRV_LLVM_Translator_unified"
+repo = "https://github.com/maleadt/SPIRV-LLVM-Translator.git"
+version = v"0.1"
+
+llvm_versions = [v"11.0.1", v"12.0.1", v"13.0.0"]
+
+# Collection of sources required to build SPIRV_LLVM_Translator
+sources = Dict(
+    v"11.0.1" => [GitSource(repo, "caa4038025861069adf857be8a5214e6857ff182")],
+    v"12.0.1" => [GitSource(repo, "3ed66fc7524d648dec5a71c5e09985d5aff5bd99")],
+    v"13.0.0" => [GitSource(repo, "776d9041e3baf120222e88842a9d441218a3f791")]
+)
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = expand_cxxstring_abis(supported_platforms(; experimental=true))
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd SPIRV-LLVM-Translator
+install_license LICENSE.TXT
+
+CMAKE_FLAGS=()
+
+# Release build for best performance
+CMAKE_FLAGS+=(-DCMAKE_BUILD_TYPE=Release)
+
+# Install things into $prefix
+CMAKE_FLAGS+=(-DCMAKE_INSTALL_PREFIX=${prefix})
+
+# Explicitly use our cmake toolchain file and tell CMake we're cross-compiling
+CMAKE_FLAGS+=(-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN})
+CMAKE_FLAGS+=(-DCMAKE_CROSSCOMPILING:BOOL=ON)
+
+# Tell CMake where LLVM is
+CMAKE_FLAGS+=(-DLLVM_DIR="${prefix}/lib/cmake/llvm")
+
+# Build the library
+CMAKE_FLAGS+=(-DBUILD_SHARED_LIBS=ON)
+
+cmake -B build -S . -GNinja ${CMAKE_FLAGS[@]}
+ninja -C build -j ${nproc} llvm-spirv install
+install -Dm755 build/tools/llvm-spirv/llvm-spirv${exeext} -t ${bindir}
+"""
+
+# The products that we will ensure are always built
+products = Product[
+    LibraryProduct(["libLLVMSPIRVLib", "LLVMSPIRVLib"], :libLLVMSPIRV, dont_dlopen = true),
+    ExecutableProduct("llvm-spirv", :llvm_spirv),
+]
+
+augment_platform_block = """
+    using Base.BinaryPlatforms
```

Comment:
Why is this needed?  Unless I'm missing something, in this snippet I don't see functions/variables/types from there

Reply:
Ah yes, good catch, it was using `BinaryPlatforms.add_tag!` before.

Reply:
Added it back for a `::Platform` type assertion (to follow the template from the Pkg syntax)